src/components/hmi_message_handler/src/mb_controller.cc

```diff
@@ -0,0 +1,479 @@
+         int position = method.find(".");
+         if(position != -1) {
+            return_string = method.substr(0, position);
+         }
+      }
+      return return_string;
+   }
+
+   bool CMessageBrokerController::addController(WebsocketSession* ws_session, std::string name){
+      bool result = false;
+      std::map <std::string, WebsocketSession*>::iterator it;
+
+      sync_primitives::AutoLock lock(mControllersListLock);
+      it = mControllersList.find(name);
+      if (it == mControllersList.end()) {
+         mControllersList.insert(std::map <std::string, WebsocketSession*>::value_type(name, ws_session));
+         result = true;
+      } else {
+         DBG_MSG(("Controller already exists!\n"));
+      }
+
+      DBG_MSG(("Count of controllers: %zu\n", mControllersList.size()));
+      return result;   
+   }
+
+   void CMessageBrokerController::deleteController(WebsocketSession* ws_session) {
+      {
+         sync_primitives::AutoLock lock(mControllersListLock);
+         std::map <std::string, WebsocketSession*>::iterator it;
+         for(it = mControllersList.begin(); it != mControllersList.end(); it++) {
+            if(it->second == ws_session) {
+               mControllersList.erase(it);
+            }
+         }
+      }
+      removeSubscribersBySession(ws_session);
+   }
+
+   void CMessageBrokerController::deleteController(std::string name) {
+      std::map <std::string, WebsocketSession*>::iterator it;
+      WebsocketSession* ws;
+      {
+         sync_primitives::AutoLock lock(mControllersListLock);
+         it = mControllersList.find(name);
+         if (it != mControllersList.end())
+         {
+            ws = it->second;
+            mControllersList.erase(it);
+         } else {
+            return;
+         }
+      }
+      removeSubscribersBySession(ws);      
+   }
+
+   void CMessageBrokerController::removeSubscribersBySession(const WebsocketSession* ws) {
+      sync_primitives::AutoLock lock(mSubscribersListLock);
+      std::multimap <std::string, WebsocketSession*>::iterator it_s = mSubscribersList.begin();
+      for (; it_s !=mSubscribersList.end(); ) {
+         if (it_s->second == ws) {
+            mSubscribersList.erase(it_s++);
+         } else {
+            ++it_s;
+         }
+      }
+   }
+
+   void CMessageBrokerController::pushRequest(Json::Value& message, WebsocketSession* ws_session) {
+      sync_primitives::AutoLock lock(mRequestListLock);
+      std::string id = message["id"].asString();
+      mRequestList.insert(std::map <std::string, WebsocketSession*>::value_type(id, ws_session));
+   }
+
+   bool CMessageBrokerController::addSubscriber(WebsocketSession* ws_session, std::string name) {
+      bool result = true;
+      sync_primitives::AutoLock lock(mSubscribersListLock);
+      std::pair<std::multimap <std::string, WebsocketSession*>::iterator, std::multimap <std::string, WebsocketSession*>::iterator> p = mSubscribersList.equal_range(name);
```

Comment:
This might be a good place to use `auto` instead of writing out the explicit type.